### PR TITLE
Issue #93 Adding basic CRC tests

### PR DESCRIPTION
**Pull request by jsliacan**
Elementary tests which check behavior of `crc`, `crc version`, `crc --help`, `crc setup`, `crc start`, `crc stop`, and `crc delete`. These are just initial/'PoC' tests, more thorough testing needed.

### CODE REVIEW

**Comment by jsliacan** [comment]
Fail is due to some (yet unused) code needing functions from clicumber which are not exposed (`executeCommand` needs to be `ExecuteCommand`, etc.). 

Also, `crc_libvirt_0.16.1.tar.gz` is expected to be (currently) in `~/Downloads`. This might be a problem.

**Comment by gbraad** [comment]
We will have a look at it and provide feedback. No worries about the 'fail'.

**Comment by gbraad** [comment]
commits are messy, and likely need to be rebased

**Comment by jsliacan** [comment]
> commits are messy, and likely need to be rebased

@gbraad , I rebased so that there are no repetitions among commits. I also changed the hardcoded name of the crc tarball in the feature file to the latest one. The tests run & pass on my setup.

**Comment by anjannath** [comment]
@jsliacan `+integration` build tag also needed in `crcsuite.go` https://github.com/code-ready/crc/pull/95/files#diff-7a2c6f6e921af901aa6dbc9ee4d87c51
Also add `godog` and `clicumber` to vendor: `make vendor` then commit those too.

**Comment by anjannath** [comment]
This looks good, but the _crc stop_ scenario is not working as expected..
```gherkin
$ make integration
[.....]
  
  Scenario: CRC stop                   # features/basic.feature:37
    When executing "crc stop" succeeds # shell.go:234 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then stdout should contain         # shell.go:288 -> github.com/code-ready/clicumber/testsuite.CommandReturnShouldContainContent
      """ Stopping "crc"...\n Machine "crc" was stopped.\n true
      
      """

  Scenario: CRC delete                   # features/basic.feature:43
    When executing "crc delete" succeeds # shell.go:234 -> github.com/code-ready/clicumber/testsuite.ExecuteCommandSucceedsOrFails
    Then stdout should contain "true"    # shell.go:284 -> github.com/code-ready/clicumber/testsuite.CommandReturnShouldContain

6 scenarios (6 passed)
26 steps (26 passed)
6m6.047359479s
ok  	github.com/code-ready/crc/test/integration	366.059s
```
From the above run all the tests seem to have passed, but from the logs the stop actually failed:
```
12:19:12.037798    info: ----- Scenario: CRC stop -----
12:19:12.037816    bash: crc stop
12:19:12.042300  stdout: crc - Local OpenShift 4.x cluster
12:19:12.062135  stdout: Stopping "crc"...
12:20:42.107499  stdout: false
12:20:42.107521  stdout: VM Failed to gracefully shutdown, try the kill command
12:20:42.108195  stdout: exitCodeOfLastCommandInShell=0
12:20:42.108311    info: ----- Scenario: CRC delete -----
12:20:42.108327    bash: crc delete
12:20:42.111176  stdout: crc - Local OpenShift 4.x cluster
12:20:42.367257  stdout: true
12:20:42.367854  stdout: exitCodeOfLastCommandInShell=0
12:20:42.367974    info: ----- Cleaning after feature -----
12:20:42.368237    info: ----- Cleaning Up -----
```

File: test/integration/features/basic.feature [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+	Scenario: CRC start
+		When executing "crc start -b ~/Downloads/crc_libvirt_v4.1.0.rc0.tar.xz" succeeds
+		Then stdout should contain "Creating VM"
+		And stdout should contain "Running"
+
+	Scenario: CRC stop
+		When executing "crc stop" succeeds
+		Then stdout should contain 
+		"""Stopping "crc"...\n Machine "crc" was stopped.\n true
```
Comment (anjannath) [inline test/integration/features/basic.feature]: @jsliacan See https://github.com/code-ready/crc/pull/95#issuecomment-489533207

File: test/integration/crcsuite/crcsuite.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	actual := sh.GetLastCmdOutput("stdout")
+	if actual != expected {
+		return fmt.Errorf("CRC state does not match %s. Expected: %s, Actual: %s.\n", expected, expected, actual)
+	}
+
+	return nil
+}
+
+// FeatureContext defines godog.Suite steps for the test suite.
+func FeatureContext(s *godog.Suite) {
```
Comment (anjannath) [inline test/integration/crcsuite/crcsuite.go]: These step definition are not used atm, but i guess in the next iteration they will be..

Comment (jsliacan) [reply test/integration/crcsuite/crcsuite.go]: Yes, that's prepared for the future (for tests which are CRC specific but use Clicumber).

**Pull request merged**